ili/utils/samplers.py

```diff
@@ -73,26 +74,39 @@ def sample(self, nsteps: int, x: np.ndarray,
             x (np.ndarray): data to evaluate the posterior at
             progress (bool, optional): whether to show progress bar.
                 Defaults to False.
+            skip_initial_state_check (bool, optional): If True, a check that 
+                the initial_state can fully explore the space will be skipped. 
+                Defaults to False.
+
+
         """
-        theta0 = np.stack([self.posterior.prior.sample()
-                          for i in range(self.num_chains)])
+        theta0 = [self.posterior.prior.sample()
+                          for i in range(self.num_chains)]
+        if not isinstance(theta0[0], np.ndarray):
+            theta0 = [t.cpu() for t in theta0]
+        theta0 = np.stack(theta0)
```

Comment:
Can't we just `.cpu()` them after stacking? They all should be on the same device in the first place.

Reply:
Good point. I will do this instead